polaris-react/src/components/IndexTable/IndexTable.module.scss

```diff
@@ -159,6 +159,10 @@ $loading-panel-height: 53px;
     cursor: auto;
   }
 
+  &.TableRow-borderless:not(:first-child) {
+    border-top: 0;
```

Comment:
Removing the border will affect the row height making it inconsistent across rows with different `divider` values. Maybe instead of removing the border it's worth making it transparent instead which would keep the same element dimensions and remove the visual border?

Reply:
1. Trying to hide borders via applying transparent border brings implications with sticky cells, [comment here](https://github.com/Shopify/polaris/pull/11676#discussion_r1520626007). 

With setting transparent color for borders  we have to:
- add styles to make borders the same colour as background for all sticky cells that do not have white background (rows with status, subheaders). Without colored borders  sticky cell has white spaces around it. I was trying to figure out how to eliminate this, but even chatGPT says  `it is likely due to the way the position: sticky property works in conjunction with table elements.`, `border-collapse: collapse;`, - didn't help as well.
- when it's feasible to add all those additional styles for sticky cells, I think it will bring implications to support those style, as developers have to take into account adding new types of rows

2. Trying to hide borders via applying border the same background color  brings implications with sticky cells for row with status, subheaders, as they all needs owerrides for sticky cells 

3. Moreover I checked how it works in Bootstrap toolkit, for example, and they also set `border: 0 ` to hide borders: https://getbootstrap.com/docs/4.1/content/tables/#borderless-table.

So seems like setting border: 0 is the solution that is easy to maintain and support. 

@tatianau do you think  ` border: 0` is still no way to go for us?